Homework_2.playground/Contents.swift

```diff
@@ -0,0 +1,98 @@
+class Stack {
+    private let linkedList = LinkedList()
+    
+    func push(element: Int) {
+        linkedList.append(element)
+    }
+    
+    func pop() -> Int? {
+        return linkedList.removeLast()
+    }
+    
+    func top() -> Int? {
+        return linkedList.last
+    }
+}
+
+class LinkedList {
+    private var head: Frame?
+    var last: Int? {
+        return head?.data
+    }
+    
+    func append(_ element: Int) {
+        head = Frame(data: element, nextFrame: head)
+    }
+    
+    func removeLast() -> Int? {
+        guard let head = head else { return nil }
+        self.head = head.next
+        return head.data
+    }
+}
+
+class Frame {
+    let data: Int
+    let next: Frame?
+    
+    init(data: Int, nextFrame: Frame?) {
+        self.data = data
+        self.next = nextFrame
+    }
+}
+
+let stack = Stack()
+
+stack.pop()
+stack.top()
+stack.push(element: 3)
+stack.push(element: 2)
+stack.push(element: 1)
+stack.top()
+stack.pop()
+stack.pop()
+stack.pop()
+stack.pop()
+stack.top()
+
+class StackStatistics: Stack {
+    private let stackOfMinElements = Stack()
+    
+    override func push(element: Int) {
+        super.push(element: element)
+        
+        guard let currentMinElement = stackOfMinElements.top(), currentMinElement < element else {
+            stackOfMinElements.push(element: element)
+            return
+        }
+        stackOfMinElements.push(element: currentMinElement)
+    }
+    
+    override func pop() -> Int? {
+        guard let number = super.pop() else { return nil }
```

Comment:
nit: You name an item in the stack as `element` earlier. Could you please explain why it's `number` here?